sat/cli/bootprep/image.py

```diff
@@ -44,26 +44,83 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def validate_unique_image_ref_names(input_images):
+    """Validate that all images have unique value for ref_name property.
+
+    Args:
+        input_images (list of sat.cli.bootprep.input.image.BaseInputImage): the
+            input images in the bootprep input instance
+
+    Raises:
+        ImageCreateError: if the images do not have unique ref_name values
+    """
+    counts_by_ref_name = Counter(image.ref_name for image in input_images if image.ref_name)
+    non_unique_ref_names = [name for name, count in counts_by_ref_name.items() if count > 1]
+    if non_unique_ref_names:
+        raise ImageCreateError(f'Images must have unique values for "ref_name" if specified.'
+                               f'Non-unique names: '
+                               f'{", ".join(non_unique_ref_names)}')
+
+
 def validate_unique_image_names(input_images):
     """Validate that the input images all have unique names
 
     Images must have unique names, so that they can be referenced by name in the
     session templates in a bootprep input file.
 
     Args:
-        input_images (list of sat.cli.bootprep.input.image.InputImage): the input
-            images in the bootprep input instance
+        input_images (list of sat.cli.bootprep.input.image.BaseInputImage): the
+            input images in the bootprep input instance
 
     Raises:
         ImageCreateError: if the images do not have unique names
     """
-    counts_by_name = Counter(image.name for image in input_images)
+    counts_by_name = Counter()
+
+    failed_name_renders = 0
+    for image in input_images:
+        try:
+            counts_by_name[image.name] += 1
+        except ImageCreateError as err:
+            failed_name_renders += 1
+            LOGGER.error(f'Failed to render name of {image}: {err}')
+
+    if failed_name_renders:
+        raise ImageCreateError(f'Failed to render {failed_name_renders} image names.')
+
     non_unique_names = [name for name, count in counts_by_name.items() if count > 1]
     if non_unique_names:
-        raise ImageCreateError('Names of images to create must be unique. Non-unique names: '
+        raise ImageCreateError(f'Names of images to create must be unique. Non-unique names: '
```

Comment:
nit: FWIW, I don't think the `f` is actually necessary since there aren't any variable substitutions on this line. See [PEP-498](https://peps.python.org/pep-0498/#concatenating-strings).

Reply:
I tend to just do `f` on all parts of concatenated strings when I have any need for it in the overall concatenated strings for two reasons:

* In case somebody adjusts the line length
* In case somebody adds a variable and doesn't realize each part isn't an f-string.
* For visual consistency with open single-quote chars being aligned